lib/logger/lib/logger.ex

```diff
@@ -573,7 +573,7 @@ defmodule Logger do
     if compare_levels(level, min_level) != :lt do
       macro_log(level, data, metadata, caller)
     else
-      :ok
+      handle_unused_variable_warnings(data)
```

Comment:
I'm wondering, maybe we should call given functions? What if someone relies on this, for example:
```elixir
Logger.debug("Done: #{cleanup()}")
```